decision-records/006-javascript-compatibility.md

```diff
@@ -0,0 +1,161 @@
+# JavaScript browser compatibility
+
+Date: 27/06/23
+
+Status: Accepted
+
+## Decision
+
+We will ship JavaScript that can parse without errors in browsers supporting `<script type="module">`.
+
+We will update our guidance to tell service teams to include the JavaScript from GOV.UK Frontend using `<script type="module">`.
+
+We will set up:
+
+- transpiling to transform modern code down to that baseline
+- linting to prevent us from using newer features and syntaxes
+
+Going forward, when we want to use a newer feature we will disable the linting rule only after checking the performance impact of the transformation.
+
+We don't believe we need to introduce any polyfills for the components that currently exist, so for now the transpilation step will not include any. We will revisit this when we want to introduce a new feature that would require polyfills.
```

Comment:
Probably remove this paragraph if it's not a decision?

Or turn it into a decision? Few things to extract:

1. We might use feature detection to guard functionality behind progressive enhancement
2. We might use polyfills to enable functionality for all `<script type="module">` users
3. We might avoid polyfills when they affect browser performance or file sizes

Reply:
Clarified that I was trying to record the fact that we'd set up Babel to only add transformations and not polyfill. I think the options for polyfilling are outlined in the browser support docs.